tests/test_signal_generator.cpp

```diff
@@ -0,0 +1,150 @@
+/**
+ * @file test_signal_generator.cpp
+ * @author Luciana Moreno (lumoreno.borges@gmail.com)
+ * @brief Unit test for the signal generator hyro package
+ * @version 0.1
+ * @date 2019-01-25
+ * 
+ * @copyright Copyright (c) 2019
+ * 
+ */
+#include <hyro/SignalGeneratorComponent.h>
+#include <hyro/DigitalConverterComponent.h>
+#include <hyro/utils/DynamicPropertyAccess.h>
+#include <hyro/test/ComponentTester.h>
+#include <gtest/gtest.h>
+
+/* Macro definition to configure the components */
+static void
+__assert_and_run_state_machine(hyro::StateMachine& sm,
+                               const hyro::ComponentConfiguration& conf,
+                               const hyro::ConnectionConfiguration& endpoint)
+{
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.init(conf));
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.start());
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.connect(endpoint));
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.check());
+}
+
+#define ASSERT_AND_RUN_STATE_MACHINE(sm, conf, endpoint) \
+  __assert_and_run_state_machine(sm, conf, endpoint)
+
+namespace hyro
+{
+namespace test
+{
+
+/**
+ * @brief Construct a new TEST object for SignalGeneratorComponent
+ * 
+ */
+TEST (TestSignalGenerator, SignalGeneratorComponentCheck)
+{
+
+  auto signal_generator_configuration = 
+      "{"
+        "outputs: {"
+           "signals: { protocol: 'api'},"
+           "fix_dynamic: {protocol: 'api'}"
+        "}" 
+      "}";
+
+
+  StateMachine signal_generator_sm(std::make_shared<SignalGeneratorComponent>("/signal_generator"_uri));
+
+  auto result = signal_generator_sm.init(ComponentConfiguration(signal_generator_configuration));
+
+  /* Teste state transitions */
+  ASSERT_EQ(Result::RESULT_OK, result);
+  result = signal_generator_sm.start();
+  ASSERT_EQ(Result::RESULT_OK, result);
+  result = signal_generator_sm.connect(ConnectionConfiguration());
+  ASSERT_EQ(Result::RESULT_OK, result);
+  result = signal_generator_sm.check();
+  ASSERT_EQ(Result::RESULT_OK, result);
+
+  /* Create a fake input channel and test connection*/
+  auto inputPos = std::make_shared<FakeInput<Signal>>("inputPos"_uri, "api", "/signal_generator/signals");
```

Comment:
Usually we define the channel URI using **snake_case** style. Please, rename **inputPos** to something like **fake_input** or **input_signal**.

Reply:
Why did you set the variable name as `inputPos`?